tests/store/test_rest_store.py

```diff
@@ -105,6 +105,19 @@ def test_requestor(self, request):
             self._verify_requests(mock_http, creds,
                                   "runs/log-metric", "POST", body)
 
+        with mock.patch('mlflow.store.rest_store.http_request_safe') as mock_http:
```

Comment:
let's add example(s) with real batches.
- a bunch of just metrics, tags and params are empty
- just tags,
- one tag, and a bunch of params
- params and tags with the same key
- metric and param batches swapped in API call, and other such error cases
- ... etc

ideally, we should have a generic test that is super comprehensive with all these options and applied to a any backend and ensure that the behavior is as intended